### ISSUE TITLE
Add pyproject.toml as a default file name option

### ISSUE THREAD

**Issue by berislavlopac** [jazzband/pip-tools#1779]
#### What's the problem this feature will solve?

Currently, `pyproject.toml` needs to be explicitly declared when running `pip-compile`:

```shell
> pip-compile
Usage: pip-compile [OPTIONS] [SRC_FILES]...
Try 'pip-compile -h' for help.

Error: Invalid value: If you do not specify an input file, the default is requirements.in or setup.py
>
> pip-compile pyproject.toml
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile pyproject.toml
#
anyio==3.6.2
    # via httpcore
(etc)
```

#### Describe the solution you'd like

Since `pip-compile` already supports two standard file formats (`requirements.in` and `setup.py`), it would make sense to add `pyproject.toml` to the list of automatically recognised formats.

**Comment by webknjaz** [comment]
Looks like `setup.cfg` would also need to be added to the list.

**Comment by berislavlopac** [comment]
It turns out that it's not necessary. I'm just preparing a PR for this, and `setup.cfg` is supposed to be accompanied either by `pyproject.toml (for PEP 517 compatible builds; see https://setuptools.pypa.io/en/latest/build_meta.html), or by `setup.py` (for legacy builds; see https://setuptools.pypa.io/en/latest/userguide/declarative_config.html). 😊

**Comment by atugushev** [comment]
Closed this as completed in #1780.
